.github/scripts/apidoc.sh

```diff
@@ -91,7 +91,7 @@ COMMIT_MSG="Update docs for $AWESOME_VERSION via Github Actions
 Last commit message:
 $LAST_COMMIT_MSG
 
-Commits: https://github.com/awesomeWM/awesome/compare/${GITHUB_BASE_REF}..${GITHUB_HEAD_REF}
+Commits: https://github.com/awesomeWM/awesome/compare/${GITHUB_BASE_REF}..${GITHUB_REF}
```

Comment:
I don't think this one will work. `GITHUB_BASE_REF` isn't set in `push` events, either.

However, based on [this](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push), GitHub seems to build this link for us (see the `compare` field). I'm not quite sure what the URL will look like, but something like this might work:

```diff
diff --git a/.github/scripts/apidoc.sh b/.github/scripts/apidoc.sh
index b341fca3..b78060c8 100755
--- a/.github/scripts/apidoc.sh
+++ b/.github/scripts/apidoc.sh
@@ -91,7 +91,7 @@ COMMIT_MSG="Update docs for $AWESOME_VERSION via Github Actions
 Last commit message:
 $LAST_COMMIT_MSG
 
-Commits: https://github.com/awesomeWM/awesome/compare/${GITHUB_BASE_REF}..${GITHUB_HEAD_REF}
+Commits: $GITHUB_COMPARE
 Build URL: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID"
 
 git commit -m "[relevant] $COMMIT_MSG"
diff --git a/.github/workflows/apidoc.yml b/.github/workflows/apidoc.yml
index a58b7e99..50c8c94d 100644
--- a/.github/workflows/apidoc.yml
+++ b/.github/workflows/apidoc.yml
@@ -151,6 +151,7 @@ jobs:
         env:
           PR_NUMBER: ${{ github.event.number }}
           APIDOC_TOKEN: ${{ secrets.AWESOME_ROBOT_TOKEN || github.token }}
+          GITHUB_COMPARE: ${{ github.event.compare }}
         run: .github/scripts/apidoc.sh
 
 # vim: filetype=yaml:expandtab:shiftwidth=2:tabstop=2
```

Reply:
To work with both `push` and `pull_request` events, it would need a more advanced logic. Not sure if that is needed.

Reply:
done

> To work with both push and pull_request events, it would need a more advanced logic. Not sure if that is needed.

The old one didn't work all that well with pull_request either. As long as it doesn't allow people to deface the main website, I am semi-ok with that. I would really like the bot to "really" work and post a comment in the PR pointing to the doc for the PR.

It did semi work, but you had to do a PR _from_ the AwesomeWM main project for it to work. If an used made a PR from their fork, the bot didn't post it's comment. It's why there is so few PR branches in the `apidoc` repo